src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs

```diff
@@ -76,40 +77,45 @@ protected override string GetEndingString(Document document, int position, Cance
             var root = tree.GetRootAsync(cancellationToken).WaitAndGetResult(cancellationToken);
             var text = tree.GetTextAsync(cancellationToken).WaitAndGetResult(cancellationToken);
 
-            var owningNode = GetOwningNode(root, position);
-            if (owningNode == null)
+            // Go through the set of owning nodes in leaf to root chain.
+            foreach (var owningNode in GetOwningNodes(root, position))
             {
-                return null;
-            }
-
-            SyntaxToken lastToken;
-            if (!TryGetLastToken(text, position, owningNode, out lastToken))
-            {
-                return null;
-            }
+                SyntaxToken lastToken;
+                if (!TryGetLastToken(text, position, owningNode, out lastToken))
+                {
+                    // If we can't get last token, there is nothing more to do, just skip
+                    // the other owning nodes and return.
+                    return null;
+                }
 
-            if (!CheckLocation(text, position, owningNode, lastToken))
-            {
-                return null;
-            }
+                if (!CheckLocation(text, position, owningNode, lastToken))
+                {
+                    // If we failed this check, we indeed got the intended owner node and
+                    // inserting line ender here would introduce errors.
+                    return null;
+                }
 
-            // so far so good. we only add semi-colon if it makes statement syntax error free
-            var semicolon = SyntaxFacts.GetText(SyntaxKind.SemicolonToken);
+                // so far so good. we only add semi-colon if it makes statement syntax error free
+                var semicolon = SyntaxFacts.GetText(SyntaxKind.SemicolonToken);
 
-            var textToParse = owningNode.NormalizeWhitespace().ToFullString() + semicolon;
+                var textToParse = owningNode.NormalizeWhitespace().ToFullString() + semicolon;
```

Comment:
in the worst case, it could be a Method or a Property and that happens only if we encounter an expression bodied member syntax.